### PR TITLE
Refactor/migrate tests to typescript

### LINKED ISSUE #387: Migrate tests to Typescript
### Overview

As a developer, let's have a consistent codebase. For this, we will try to use typescript as much as possible. 

### Action Items

- [ ] Migrate existing jest tests from js to tsx if not already.

### Resources/Instructions

[Resources](https://hackforla.github.io/CivicTechJobs/resources/)

### PR DESCRIPTION
**Pull request by bennyv8**
Resolves #387 

### Changes

- Renamed components test .js files to .tsx 
- Revised the ButtonAnchorProps interface to address cases of absent href property (e.g., when it's used as a ButtonButtonProp).
- Removed awaits as screen.QueryByRole and screen.getDisplayValue are synchronous

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "CivicTechJobs",
+  "lockfileVersion": 3,
+  "requires": true,
+  "packages": {}
+}
```
Comment (MattPereira) [inline package-lock.json:6]: Looks like you may have accidentally ran `npm install` the root folder which added this file

**Review by MattPereira — CHANGES_REQUESTED** [review]
The converted tests are all passing and great job explaining your changes! :+1: 

Just need remove the errant `package.json` file in the root and then you'll be ready to merge

File: package-lock.json (hunk 1 above)
Comment (bennyv8) [reply package-lock.json]: Oops, removed

**Review by MattPereira — APPROVED** [review]

**Review by bzzz-coding — APPROVED** [review]
@bennyv8 Thanks for working on this issue ✨! All tests passed and the code looks clean. Please go ahead and merge your PR. 🚀

**Comment by MattPereira** [comment]
trying to re-trigger status checks that are indefinitely hanging

<details><summary>See screenshot</summary>

image


</details>

**Pull request merged**